pkg/ruler/ruler.go

```diff
@@ -891,7 +1080,7 @@ func (r *Ruler) getShardedRules(ctx context.Context, userID string, rulesRequest
 		ring = r.ring.ShuffleShard(userID, shardSize)
 	}
 
-	rulers, err := ring.GetReplicationSetForOperation(ListRuleRingOp)
+	rulers, failedZones, err := ring.GetReplicationSetForOperationWithNoQuorum(ListRuleRingOp)
```

Comment:
If we are not doing `mergeGroupStateDesc` that would mean that the replication factor is set to 1 (or else there will be duplicate rules in the response). In this case whether we require quorum or not the resulting ReplicationSet is the same because `GetReplicationSetForOperation` and `GetReplicationSetForOperationWithNoQuorum` behaves the same way if replication factor is 1.